src/intrinsics/node/contextify.js

```diff
@@ -0,0 +1,404 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import invariant from "../../invariant.js";
+import { Realm, ExecutionContext } from "../../realm.js";
+import { AbruptCompletion, ThrowCompletion } from "../../completions.js";
+import { Value, ConcreteValue, BooleanValue, EmptyValue, NativeFunctionValue, ObjectValue, StringValue, UndefinedValue } from "../../values/index.js";
+import { DefinePropertyOrThrow, Set, Get, GetValue, GetFunctionRealm, ObjectCreate, ToBoolean, ToInteger, ToString } from "../../methods/index.js";
+import parse from "../../utils/parse.js";
+import type { BabelNodeFile } from "babel-types";
+
+// TODO: This creates a strong dependency on babel and its transforms even
+// outside of devDependencies which is unfortunate. Get rid of this once classes
+// and destructuring is fully implemented.
+import { transform as babelTransform } from "babel-core";
+
+// Hook for transpiling
+function transform(code : string, filename : string) : string {
+  let patchedCode = code.replace(
+    // Work around the fact that Babel classes can't extend natives.
+    /class FastBuffer extends Uint8Array {\s+constructor\(arg1, arg2, arg3\) {\s+super\(arg1, arg2, arg3\);\s+}\s+}/g,
+    'function FastBuffer(arg1, arg2, arg3) {\n' +
+    '  var self = new Uint8Array(arg1, arg2, arg3);\n' +
+    '  Object.setPrototypeOf(self, FastBuffer.prototype);\n' +
+    '  return self;\n' +
+    '}; Object.setPrototypeOf(FastBuffer, Uint8Array); Object.setPrototypeOf(FastBuffer.prototype, Uint8Array.prototype);'
+  );
+  let transformedCode = babelTransform(patchedCode, {
+    plugins: [
+      // Prepack doesn't support classes or destructuring yet.
+      'transform-es2015-classes',
+      'transform-es2015-destructuring',
+      'transform-es2015-parameters',
+    ],
+    retainLines: true,
+  });
```

Comment:
Note that this depends on a "devDependency" atm as a hack. I didn't want to add it as a hard dependency since this is still experimental and we'll just implement these directly eventually.